openfisca_core/tests/test_countries.py

```diff
@@ -256,3 +262,20 @@ def test_non_existing_variable():
         ).new_simulation()
 
     simulation.calculate('non_existent_variable', 2013)
+
+
+def test_calculate_variable_with_wrong_period_unit():
+    simulation = tax_benefit_system.new_scenario().init_single_entity(
+        period = 2013,
+        parent1 = dict(
+            salaire_brut = 4000,
+            ),
+        ).new_simulation()
+
+    with assert_raises(ValueError) as error:
+        simulation.calculate('rsa', 2013)
+
+    error_message = str(error.exception)
+    expected_error_message = 'Computation requested with wrong period unit for variable rsa (year instead of month)'
```

Comment:
Auto-review : testing the exact message of the error is too much. Maybe check that it contains some key words, or just check for the type of expression.